common.mk

```diff
@@ -1,50 +1,38 @@
-ROCKETCHIP_DIR=$(base_dir)/generators/rocket-chip
-TESTCHIP_DIR = $(base_dir)/generators/testchipip
-
-SCALA_VERSION=2.12.4
-SCALA_VERSION_MAJOR=$(basename $(SCALA_VERSION))
-
-SBT ?= java -Xmx2G -Xss8M -XX:MaxPermSize=256M -jar $(ROCKETCHIP_DIR)/sbt-launch.jar ++$(SCALA_VERSION)
-
+#########################################################################################
+# set default shell for make
+#########################################################################################
+SHELL=/bin/bash
+
+#########################################################################################
+# variables to get all *.scala files
+#########################################################################################
 lookup_scala_srcs = $(shell find $(1)/ -iname "*.scala" 2> /dev/null)
 
 PACKAGES=rocket-chip testchipip
 SCALA_SOURCES=$(foreach pkg,$(PACKAGES),$(call lookup_scala_srcs,$(base_dir)/generators/$(pkg)/src/main/scala)) $(call lookup_scala_srcs,$(base_dir)/src/main/scala)
 
+#########################################################################################
+# rocket and testchipip classes
+#########################################################################################
 ROCKET_CLASSES ?= "$(ROCKETCHIP_DIR)/target/scala-$(SCALA_VERSION_MAJOR)/classes:$(ROCKETCHIP_DIR)/chisel3/target/scala-$(SCALA_VERSION_MAJOR)/*"
 TESTCHIPIP_CLASSES ?= "$(TESTCHIP_DIR)/target/scala-$(SCALA_VERSION_MAJOR)/classes"
+
+#########################################################################################
+# jar creation variables and rules
+#########################################################################################
 FIRRTL_JAR ?= $(ROCKETCHIP_DIR)/lib/firrtl.jar
 
+# this should match whatever the commonSettings version is in build.sbt
+BARSTOOLS_VER=1.0
```

Comment:
not a big deal, but is there a specific reason to keep the version around if it's just whatever the build.sbt assembles? Could we just remove the suffix as a final stage of assembly?

Reply:
@jwright6323 Is there a reason that you did this?